Fix:

error: incomplete definition of type 'struct png_struct_def'
--- lib/pngimage.c.orig	2006-04-21 00:48:54.000000000 -0500
+++ lib/pngimage.c	2021-01-23 02:31:47.000000000 -0600
@@ -35,7 +35,7 @@
 	return NULL;
     }
 
-    if (setjmp(png_ptr->jmpbuf)) {
+    if (setjmp(png_jmpbuf(png_ptr))) {
 	png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
 	RbImage_setError("setjmp returns error condition");
 	return NULL;
@@ -101,7 +101,7 @@
 	return NULL;
     }
 
-    if (setjmp(png_ptr->jmpbuf)) {
+    if (setjmp(png_jmpbuf(png_ptr))) {
 	png_destroy_write_struct(&png_ptr, &info_ptr);
 	RbImage_setError("setjmp returns error condition");
 	return NULL;
